library/tests/integration/questions_router_test.rs

```diff
@@ -0,0 +1,325 @@
+        .json(&updated_question)
+        .reply(&routers)
+        .await;
+
+    assert_eq!(
+        put_resp.status(),
+        StatusCode::OK,
+        "Failed to update the created question"
+    );
+
+    // Test GET /questions to get the updated question
+    let get_updated_resp = request()
+        .method("GET")
+        .path(format!("/questions/{}", &raw_question_id.clone()).as_str())
+        .reply(&routers)
+        .await;
+
+    assert_eq!(
+        get_updated_resp.status(),
+        StatusCode::OK,
+        "Failed to get the updated question"
+    );
+
+    // Deserialize the response body into a Value
+    let updated_response_body: Value = serde_json::from_slice(&get_updated_resp.body())
+        .expect("Failed to deserialize response body");
+    let get_updated_question: QuestionEntity = serde_json::from_value(updated_response_body)
+        .expect("Failed to deserialize response body into QuestionEntity");
+
+    assert_eq!(
+        get_updated_question, updated_question,
+        "The updated question is not as expected"
+    );
+
+    // Test DELETE /questions to delete the created question
+    let delete_resp = request()
+        .method("DELETE")
+        .path(format!("/questions/{}", &raw_question_id.clone()).as_str())
+        .reply(&routers)
+        .await;
+
+    assert_eq!(
+        delete_resp.status(),
+        StatusCode::OK,
+        "Failed to delete the created question"
+    );
+
+    // Test GET /questions to get the deleted question
+    let get_deleted_resp = request()
+        .method("GET")
+        .path(format!("/questions/{}", &raw_question_id.clone()).as_str())
+        .reply(&routers)
+        .await;
+
+    assert_eq!(
+        get_deleted_resp.status(),
+        StatusCode::NOT_FOUND,
+        "The deleted question still exists"
+    );
+}
+
+struct DatabaseConfig {
+    url: String,
+    max_size: usize,
+}
+
+#[tokio::test]
+async fn questions_router_postgres_db_test() {
+    use deadpool_diesel::{
+        postgres::{Pool, Runtime},
+        Manager,
+    };
+    use diesel::dsl::sql_query;
+    use diesel::RunQueryDsl;
+    use rand::Rng;
+    use testcontainers_modules::{postgres::Postgres, testcontainers::clients::Cli};
+
+    // Set up a postgres database question port for testing
+    let docker = Cli::default();
+    let postgres_instance = docker.run(Postgres::default());
+
+    let database_url = String::from(format!(
+        "postgres://postgres:postgres@127.0.0.1:{}/postgres",
+        postgres_instance.get_host_port_ipv4(5432)
+    ));
+    let database_config = DatabaseConfig {
+        url: database_url.clone(),
+        max_size: 10,
+    };
+
+    let manager = Manager::new(database_config.url, Runtime::Tokio1);
+    let db_pool = Pool::builder(manager)
+        .max_size(database_config.max_size)
+        .build()
+        .unwrap();
+
+    // Create the "questions" table in the postgres database if not existed
+    let conn = db_pool.get().await.unwrap();
```

Comment:
check the diesel code and run the migration from using the actual migration folder